<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sliding MxN Puzzle</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout">
     <item row="3" column="2" colspan="2">
      <widget class="Canvas" name="canvas" native="true"/>
     </item>
     <item row="4" column="3">
      <widget class="QSpinBox" name="gridSizeY">
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>3</number>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QSpinBox" name="gridSizeX">
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="value">
        <number>3</number>
       </property>
      </widget>
     </item>
     <item row="7" column="2">
      <widget class="QLabel" name="time">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string>Time</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="7" column="3">
      <widget class="QLabel" name="ops">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string>Operations</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="6" column="2" colspan="2">
      <widget class="QPushButton" name="puzzleSolve">
       <property name="text">
        <string>Solve Puzzle</string>
       </property>
      </widget>
     </item>
     <item row="5" column="2" colspan="2">
      <widget class="QComboBox" name="solvingMethod">
       <item>
        <property name="text">
         <string>Breadth-First Search</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>A*</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QWidget</extends>
   <header>canvas.h</header>
   <container>1</container>
   <slots>
    <slot>setGridSizeX(int)</slot>
    <slot>setGridSizeY(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>gridSizeX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>canvas</receiver>
   <slot>setGridSizeX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>74</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gridSizeY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>canvas</receiver>
   <slot>setGridSizeY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>211</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
